asetypes/nullableTypes.go

```diff
@@ -0,0 +1,132 @@
+// SPDX-FileCopyrightText: 2021 SAP SE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package asetypes
+
+import (
+	"database/sql/driver"
+	"fmt"
+)
+
+// TODO Add nullable datatypes, e.g.
+// asetypes.NullDecimal
+// asetypes.NullBinary
+// ...
+//
+// For orientation:
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/database/sql/sql.go;l=175
+
+type NullDecimal struct {
+	Dec   *Decimal
+	Valid bool
+}
+
+// Scan implements the scanner interface.
+// TODO: Discuss if replacements of sql.convertAssign-functionalities
+// should be placed in a custom asetypes.convertAssign method, see
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/database/sql/convert.go;l=210
+func (nd *NullDecimal) Scan(value interface{}) error {
+	// Replaces sql.convertAssign-functionality of checking
+	// nil-destination
+	if nd == nil {
+		return fmt.Errorf("destination pointer is nil")
+	}
+
+	if value == nil {
+		nd.Dec, nd.Valid = &Decimal{}, false
+		return nil
+	}
+
+	// Replaces sql.convertAssign-functionality of checking type
+	dec, ok := value.(*Decimal)
+	if !ok {
+		return fmt.Errorf("asetypes: %T is not of type decimal", value)
+	}
+	// Replaces sql.convertAssign-functionality of setting value
+	nd.Dec = dec
+
+	if nd.Dec.i == nil {
+		nd.Valid = false
+		return nil
+	}
+
+	nd.Valid = true
+
+	return nil
+}
+
+// Value implements the driver Valuer interface.
+func (nd NullDecimal) Value() (driver.Value, error) {
+	if !nd.Valid {
+		return nil, nil
+	}
+	return nd.Dec, nil
+}
+
+type NullBinary struct {
+	ByteSlice []byte
+	Valid     bool
+}
+
+// Scan implements the scanner interface.
+// TODO: Discuss if replacements of sql.convertAssign-functionalities
+// should be placed in a custom asetypes.convertAssign method, see
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/database/sql/convert.go;l=210
+func (nb *NullBinary) Scan(value interface{}) error {
+	// Replaces sql.convertAssign-functionality of checking
+	// nil-destination
+	if nb == nil {
+		return fmt.Errorf("destination pointer is nil")
+	}
+
+	if value == nil {
+		nb.ByteSlice, nb.Valid = []byte{}, false
```

Comment:
```suggestion
		nb.ByteSlice = []byte{}
		nb.Valid = false
```

Multiple assignments may work like that but they're going against KISS.